app/models/task.rb

```diff
@@ -32,6 +35,7 @@ class Task < ApplicationRecord
   accepts_nested_attributes_for :tests, allow_destroy: true
   accepts_nested_attributes_for :model_solutions, allow_destroy: true
   accepts_nested_attributes_for :group_tasks, allow_destroy: true
+  accepts_nested_attributes_for :labels
```

Comment:
Here, we are creating new labels indirectly. We should probably also consider what happens if a label is no longer referred by any task. Currently, we are still keeping it. Is this desired ("yes" is a valid response) or should we clean unused labels?

Reply:
Since users can always create new labels when needed it is probably best to delete the unused ones.

Reply:
Yes, this probably would be desirable. The only downside I can think of is a change in the color -- since every new label gets a new random color.

Reply:
We could try to generate the color from the string. Either something like
`Digest::MD5.hexdigest(label)[0..5]`


alternativly we use a hash to generate a reproduceable number and pick a color from the COLORS-array in the label-model (that way we would have more control of the available colors)
`Digest::MD5.hexdigest(label).to_i(16) % 50`

Reply:
I went with the second solution because it sticks to the pre-defined colors.